### PR TITLE
Update Mobile New Dev Docs

### LINKED ISSUE #2483: Mobile Dev Setup Docs Need Updating
There are several issues with the mobile dev env setup docs at `packages/mobile/README.md`

* Incomplete documentation for running in a local environment
* Incomplete documentation for running in Android Studio or Xcode

### PR DESCRIPTION
**Pull request by adrastaea**
### Pull Request Checklist

- [x] I have linked this PR to a related GitHub issue.
- [x] I have added a description of the change (and Github issue number, if any) to the root [CHANGELOG.md](https://github.com/TryQuiet/quiet/blob/develop/CHANGELOG.md).

### (Optional) Mobile checklist

Please ensure you completed the following checks if you did any changes to the mobile package:

- [ ] I have run e2e tests for mobile
- [ ] I have updated base screenshots for visual regression tests

### CODE REVIEW

File: packages/mobile/README.md [hunk 1]
```diff
@@ -6,46 +6,102 @@ Quiet Mobile is a React Native app for Android and iOS that shares a Node.js [ba
...
+
+    For example, on macOS with Android Studio installed in the default location,
+    Add the following line to your `~/.bash_profile` or `~/.zprofile` file,
+    ```bash
+    export JAVA_HOME="/Applications/Android Studio.app/Contents/jbr/Contents/Home"
+    ```
+
+### Local development
+
+Follow the steps in [React Native Development Environment](https://reactnative.dev/docs/environment-setup) to set up your development environment.
```
Comment (leblowl) [inline packages/mobile/README.md]: Why is this necessary?

**Review by leblowl — COMMENTED** [review]
Looks good, just have one question

File: packages/mobile/README.md (hunk 1 above)
Comment (adrastaea) [reply packages/mobile/README.md]: This includes instructions for installing necessary dependencies like `watchman` , Android Studio, Xcode, and setup of those. It was critical for my setup.

**Review by leblowl — APPROVED** [review]
Sorry, thought I approved this. Thanks

File: packages/mobile/README.md (hunk 1 above)
Comment (leblowl) [reply packages/mobile/README.md]: Ok, I think one thing that confused me was that when I click that link it's talking about Expo and I don't think we use that, but I'm not even really sure what Expo is
